ProcessMaker/processmaker
# Add revised clear-compiled artisan command

## Issue
The command `php artisan clear-compiled` artisan command is run (on the command line or programmatically) it flushes the package manifest.

This is an issue since the licensed packages are loaded based on the package manifest and if that file is cleared, it will attempt to load all installed packages, including custom ones, which triggers a fatal error.

## Reproduction Steps
1. On the most recent build with all enterprise and custom packages installed, run the `php artisan license:update {licenseFile}` command and ensure it runs successfully
2. Attempt to run the upgrade migrations (be sure they have not been run yet) with `php artisan upgrade`
3. Notice the failure: `{"errors":{"username":["cURL error 3: URL rejected: No host part in the URL (see https:\/\/curl.haxx.se\/libcurl\/c\/libcurl-errors.html) for \/api\/1.0\/\/integratingpm3onpm4\/user?username=admin"]}}`

This happens because one of the custom packages has a `UserObserver` listening for user activities and is triggered by one of the upgrade migrations updating user information. The `UserObserver` looks for specific environment variables to be set and if they’re not found, it calls `die()` which immediately kills the process.

The greater problem is that the custom package’s service provider was being loaded in the first place. There  are two programmatic calls to the `clear-compiled` artisan command, one in an upgrade migration and one in package-cdata.

## Solution
~~Create a customized version of the php artisan clear-compiled command programmatically and run `php artisan package:discover` to ensure it is rebuilt before continuing.~~

Listen for clear-compiled and optimize:clear commands and return package:discover

## How to Test
- Run through the replication steps (be sure the upgrade migrations are reset) and ensure they are successful.
- Run the `php artisan clear-compiled` command and check through the `bootstrap/cache/packages.php` file to ensure it contains only licensed packages.

## Related Tickets & Packages
- [FOUR-12669](https://processmaker.atlassian.net/browse/FOUR-12669)

## Code Review Checklist


[FOUR-12669]: https://processmaker.atlassian.net/browse/FOUR-12669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
